src/Nethermind/Nethermind.Blockchain/KnownChainSizes.cs

```diff
@@ -2,46 +2,95 @@
         }
 
         /// <summary>
         /// Size in bytes, daily growth rate and the date of manual update
         /// </summary>
-        public static Dictionary<ulong, SizeInfo> ChainSize = new()
+        public static IChainEstimations CreateChainSizeInfo(ulong chainId)
         {
-            { BlockchainIds.Goerli, new SizeInfo(8490.MB(), 15.MB(), new DateTime(2021, 12, 7)) },
-            { BlockchainIds.Rinkeby, new SizeInfo(34700.MB(), 20.MB(), new DateTime(2021, 12, 7)) },
-            { BlockchainIds.Ropsten, new SizeInfo(35900.MB(), 25.MB(), new DateTime(2021, 12, 7)) },
-            { BlockchainIds.Mainnet, new SizeInfo(90000.MB(), 70.MB(), new DateTime(2022, 04, 7)) },
-            { BlockchainIds.Gnosis, new SizeInfo(18000.MB(), 48.MB(), new DateTime(2021, 12, 7)) },
-            { BlockchainIds.EnergyWeb, new SizeInfo(15300.MB(), 15.MB(), new DateTime(2021, 12, 7)) },
-            { BlockchainIds.Volta, new SizeInfo(17500.MB(), 10.MB(), new DateTime(2021, 11, 7)) },
-            { BlockchainIds.PoaCore, new SizeInfo(13900.MB(), 4.MB(), new DateTime(2021, 12, 7)) },
-        };
+            return chainId switch
+            {
+                BlockchainIds.Goerli => new ChainEstimations(
+                    new LinearExtrapolation(8490.MB(), 15.MB(), new DateTime(2021, 12, 7))),
+                BlockchainIds.Rinkeby => new ChainEstimations(
+                    new LinearExtrapolation(34700.MB(), 20.MB(), new DateTime(2021, 12, 7))),
+                BlockchainIds.Ropsten => new ChainEstimations(
+                    new LinearExtrapolation(35900.MB(), 25.MB(), new DateTime(2021, 12, 7))),
+                BlockchainIds.Mainnet => new ChainEstimations(
+                    new LinearExtrapolation(90000.MB(), 70.MB(), new DateTime(2022, 04, 7)),
+                    new LinearExtrapolation(163.GB(), new DateTime(2023, 4, 11, 19, 49, 0),
+                        177653453177, new DateTime(2023, 04, 29, 00, 50, 0))),
+                BlockchainIds.Gnosis => new ChainEstimations(
+                    new LinearExtrapolation(18000.MB(), 48.MB(), new DateTime(2021, 12, 7))),
+                BlockchainIds.EnergyWeb => new ChainEstimations(
+                    new LinearExtrapolation(15300.MB(), 15.MB(), new DateTime(2021, 12, 7))),
+                BlockchainIds.Volta => new ChainEstimations(
+                    new LinearExtrapolation(17500.MB(), 10.MB(), new DateTime(2021, 11, 7))),
+                BlockchainIds.PoaCore => new ChainEstimations(
+                    new LinearExtrapolation(13900.MB(), 4.MB(), new DateTime(2021, 12, 7))),
+                BlockchainIds.Sepolia => new ChainEstimations(null,
+                    new LinearExtrapolation(2457.MB(), new DateTime(2023, 04, 9, 21, 11, 0),
+                        3699505976, new DateTime(2023, 04, 28, 20, 18, 0))),
+                _ => UnknownChain.Instance
+            };
```

Comment:
@kamilchodola can you update chain sizes post newest tests?

Reply:
We have problems with chain sizes, they are very unstable. I'm working on it right now

Reply:
I guess we should do that a bit in similar way as we had with Pivots.
Tests will store data about chain size and then action will fetch them and apply in code.
Would be worth to read it from json file rather than hardcoding